src/test/groovy/de/iteratec/osm/result/PerformanceAspectSpec.groovy

```diff
@@ -0,0 +1,53 @@
+package de.iteratec.osm.result
+
+import de.iteratec.osm.csi.Page
+import de.iteratec.osm.measurement.schedule.JobGroup
+import grails.buildtestdata.BuildDataTest
+import grails.buildtestdata.mixin.Build
+import spock.lang.Specification
+
+@Build([JobGroup, Page, PerformanceAspect])
+class PerformanceAspectSpec extends Specification implements BuildDataTest {
+    Page page
+    JobGroup jobGroup
+    SelectedMeasurand metric
+
+    def setup() {
+        page = Page.build()
+        jobGroup = JobGroup.build()
+        metric = new SelectedMeasurand(Measurand.DOC_COMPLETE_TIME.name(), CachedView.CACHED)
+    }
+
+    def cleanup() {
+    }
+
+    void "ensure duplicates are not allowed"() {
+        given: "there is a performance aspect"
+        PerformanceAspect.build(
+                page: page,
+                jobGroup: jobGroup,
+                performanceAspectType: PerformanceAspectType.PAGE_CONSTRUCTION_STARTED,
+                metric: metric).save(flush:true)
+
+        when: "a duplicate to the first aspect is created"
+        PerformanceAspect duplicateAspect = new PerformanceAspect(page: page, jobGroup: jobGroup, performanceAspectType: PerformanceAspectType.PAGE_CONSTRUCTION_STARTED, metric: metric)
+
+        then: "the duplicate's validation fails"
+        !duplicateAspect.validate()
```

Comment:
Please parametrize this test with a where block. That would make it more clear that there may be multiple aspects for a page/jobGroup combination very well. Just not for the same aspect type:

page | jobGroup | performanceAspectType | metric || valid

Reply:
Fix is on child branch of this branch which is also in review